assets/js/query_params.js

```diff
@@ -1,21 +1,58 @@
 import { isLocalStorageAvailable } from './storage_availability.js';
 
 const ALLOWED_KEYS = Object.freeze(['agency', 'integration']);
+const CACHE_KEY = 'lastParamsUpdate';
+const SEVEN_DAYS_IN_MILLISECONDS = 604800000;
+const CACHE_DURATION = SEVEN_DAYS_IN_MILLISECONDS;
+
+const currentTimeMilliseconds = () => Date.now();
+
+function hasAllowedParams(urlParams, allowedKeys) {
+  return allowedKeys.some((key) => urlParams.has(key));
+}
+
+function isCacheExpired(lastUpdate, currentTime) {
+  const elapsedTime = currentTime - lastUpdate;
+  return !!lastUpdate && elapsedTime > CACHE_DURATION;
+}
+
+function clearStoredParams(allowedKeys, urlParams) {
+  const currentTime = currentTimeMilliseconds();
+  const lastUpdate = localStorage.getItem(CACHE_KEY);
+
+  if (hasAllowedParams(urlParams, allowedKeys) || isCacheExpired(lastUpdate, currentTime)) {
+    allowedKeys.forEach((key) => localStorage.removeItem(key));
+    localStorage.removeItem(CACHE_KEY);
+
+    return true;
+  }
+
+  return false;
```

Comment:
Doesn't appear we're doing anything with the return value?

```suggestion
  }
```

Reply:
No but I thought it was nice to have :) Removed.